### ISSUE TITLE
Button: throws runtime error when passed with a dynamic string as a child 

### ISSUE THREAD

**Issue by Harshit369** [innovaccer/design-system#262]
## Description

Button children are types as `string`. But when a condition is passed inside it throws an error at runtime

### Steps to reproduce

`<Button> {true ? 'a' : 'b' } </Button>`

![image](https://user-images.githubusercontent.com/7912702/89284546-91412880-d66c-11ea-934f-4bacbd71219d.png)

**Comment by aditya-kumawat** [comment]
It throws error as the children type is not string if you are having white-space before/after `{` and `}`.
This works: 
```
<Button>{true ? 'a' : 'b' }</Button>
```
OR
```
<Button>
   {true ? 'a' : 'b' }
</Button>
```

**Comment by Harshit369** [comment]
> It throws error as the children type is not string if you are having white-space before/after `{` and `}`.
> This works:
> 
> ```
> <Button>{true ? 'a' : 'b' }</Button>
> ```
> 
> OR
> 
> ```
> <Button>
>    {true ? 'a' : 'b' }
> </Button>
> ```

But isn't that a catch.. is there a reason we are not allowing `React.ReactChildren` as children?

**Comment by Harshit369** [comment]
the design demands a button like this: 
![image](https://user-images.githubusercontent.com/7912702/89290887-2b0dd300-d677-11ea-93f4-40d6a3278593.png)
i was using `Custom &nbsp;&nbsp;&nbsp;` to suffice for the margin between icon and text? is there a workaround for this that I'm not aware of

**Comment by aditya-kumawat** [comment]
Are you using this Button for Dropdown trigger or is it a normal Button?
If it's a normal Button, you need to discuss it with design team that it's not as per design-guidelines.

**Comment by Harshit369** [comment]
Acts as a trigger for range picker.
![image](https://user-images.githubusercontent.com/7912702/89293746-09fbb100-d67c-11ea-8a43-0e8ce586123b.png)

**Comment by satyamyadav** [comment]
Is resolved by #270
